### PR TITLE
Fixes #2707: Use Files.newInputStream instead of FileInputStream directly

### LINKED ISSUE #2707: web-fragment.xml support introduced in 0.16.0 is broken
`ZipPath` does not support `toFile()` so this code won't work:
https://github.com/quarkusio/quarkus/blob/master/extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/WebXmlParsingBuildStep.java#L109
and throw:
```
Caused by: java.lang.UnsupportedOperationException
at com.sun.nio.zipfs.ZipPath.toFile(ZipPath.java:597)
at io.quarkus.undertow.deployment.WebXmlParsingBuildStep.parseWebFragments(WebXmlParsingBuildStep.java:109)
at io.quarkus.undertow.deployment.WebXmlParsingBuildStep.createWebMetadata(WebXmlParsingBuildStep.java:88)
```

This is annoying because as long as you have a jar with a `web-fragment.xml`, you're not able to compile your application anymore.

Note: we should probably have a test for that. We already have a few tests with external jars (for JPA entities). It would be safer.

### PR DESCRIPTION
**Pull request by gastaldi**

### CODE REVIEW

**Comment by gsmet** [comment]
@gastaldi any chance you could work on a test?

That's what prevented me from fixing this one right away :).

**Comment by geoand** [comment]
I was doing the same thing, but it's not enough unfortunately :) since it will blow up here: https://github.com/quarkusio/quarkus/blob/b046194de00415ebcf53207919b3fd805c18bf74/extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/WebXmlParsingBuildStep.java#L116

**Comment by gastaldi** [comment]
@gsmet yeah, I'm going through the ShrinkWrap API to figure out how to do that.
@geoand you seem to be in a more advanced stage than me, feel free to reject/close this PR if you find a solution faster than me :)

**Comment by geoand** [comment]
@gastaldi no worries, go ahead since I am sure you'll figure out the testing part before I do, so feel free to use the info I posted above about the problem I mentioned :)

**Comment by gsmet** [comment]
@geoand yes, exactly. It doesn't need to be a separate one if all the Undertow tests are in the main integration tests.

As stated in the issue, we already do that for Hibernate ORM with the `common-jpa-entities` artifact.

**Comment by geoand** [comment]
Then we just need to find a jar that contains `web-fragment.xml`. The one I tested unfortunately is myfaces from `master` so it's not released. If someone has a jar in mind, that would be great.

**Comment by gsmet** [comment]
You two are so cute :).

**Comment by gastaldi** [comment]
@geoand my idea is to generate this JAR using ShrinkWrap and add it to the JAR that's managed by Quarkus.

**Comment by geoand** [comment]
@gastaldi I'm looking forward to seeing that in action :wink:

**Comment by gsmet** [comment]
Not preventing you to do it if you want to have fun but we really need to be as close to real life as possible.

At first, the external jar approach had my preference for this very reason.

**Comment by geoand** [comment]
@gastaldi should we add the `coding in progress` label?

File: integration-tests/main/pom.xml [hunk 1]
```diff
@@ -163,6 +163,12 @@
             <groupId>org.webjars</groupId>
             <artifactId>bootstrap</artifactId>
         </dependency>
+        <dependency>
```
Comment (geoand) [inline integration-tests/main/pom.xml]: Perhaps there should be a comment explaining why this dependency is added?

File: extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/WebXmlParsingBuildStep.java [hunk 2]
```diff
@@ -101,19 +99,19 @@ WebMetadataBuildItem createWebMetadata(ApplicationArchivesBuildItem applicationA
     /**
      * parse web-fragment.xml
      */
-    public Map<String, WebFragmentMetaData> parseWebFragments(ApplicationArchivesBuildItem applicationArchivesBuildItem) {
-        Map<String, WebFragmentMetaData> webFragments = new HashMap<>();
+    private Set<WebFragmentMetaData> parseWebFragments(ApplicationArchivesBuildItem applicationArchivesBuildItem) {
+        Set<WebFragmentMetaData> webFragments = new HashSet<>();
```
Comment (gsmet) [inline extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/WebXmlParsingBuildStep.java]: Let's use a `Linked` one here.

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks mostly good to me. Could you adjust the type of the `Set` and squash everything into one commit?

Thanks!

**Review by gsmet — APPROVED** [review]

**Pull request merged**